### PR TITLE
fix(coinbase) - add usePrivate option

### LINKED ISSUE #22571: Coinbase - provide an option to use the private methods instead of public methods to reduce rate limit errors
### Preliminary Checks

- [X] I have already searched for [existing issues](https://github.com/ccxt/ccxt/issues) and confirmed that this issue is not a duplicate

### Is your feature request related to a problem? Please describe

Using the public methods have a lower rate limit and result in Too Many Request errors from coinbase.

### Describe the solution you'd like

I've replaced all calls to PublicGetBrokerageMarketProducts with PrivateGetBrokerageProducts as a patch,  but it's not a great solution for all users.  Better if this was controlled via an Option in the exchange class.

sed -i "s/v3PublicGetBrokerageMarketProducts/v3PrivateGetBrokerageProducts/g" ~/.local/lib/python3.8/site-packages/ccxt/coinbase.py

### Describe alternatives you've considered

_No response_

### PR DESCRIPTION
**Pull request by ttodua**
fix #22571

### CODE REVIEW

File: ts/src/coinbase.ts [hunk 1]
```diff
@@ -344,6 +344,7 @@ export default class coinbase extends Exchange {
                 'CGLD': 'CELO',
             },
             'options': {
+                'usePrivateEndpointsIfApikeysPresent': false,
```
Comment (carlosmiei) [inline ts/src/coinbase.ts:347]: I wouldn't do it like this. This is nonstandard.

We should read an option`method` and maybe `private` using handleOptionAndParams (and add that to the docs). then inside each method we use that to decide if we use one or the other endpoint

Comment (jtashiro) [reply ts/src/coinbase.ts:347]: Take a look at this commit a1fe0b95020254445f5919af7eead84c6b109b2b which includes a test for credentials and switches from public to private API calls ... this is the pattern I was suggesting earlier.  Perhap I'm not reading this correctly?  Decide if this is non-standard approach or not?
+        if (this.checkRequiredCredentials (false)) {
+            spotUnresolvedPromises.push (this.v3PrivateGetBrokerageTransactionSummary (params));
+        }

**Comment by ttodua** [comment]
i decided to refrain from adding static tests, as they involve more changes (changes in test.ts too) to handle apiKey & secret fields, but i have pending 'tests reorg PR' so to avoid much conflicts, i'll wait that PR at first.

File: ts/src/coinbase.ts (hunk 1 above)
Comment (ttodua) [reply ts/src/coinbase.ts:347]: @carlosmiei so do you suggest that we have to add `private: true` in 5 methods under `options` one-by-one?  also `private` is not obviously intuitive, we might need more expressive key name.

Comment (carlosmiei) [reply ts/src/coinbase.ts:347]: you don't need to add one by one in options, but we can have a global option like `usePrivateEndpoint` and inside each method you read the value using `handleOptionAndParams`

Comment (ttodua) [reply ts/src/coinbase.ts:347]: > you don't need to add one by one in options, but we can have a global option like `usePrivateEndpoint` and inside each method you read the value using `handleOptionAndParams`

but is not that what is being done in this PR?

Comment (carlosmiei) [reply ts/src/coinbase.ts:347]: kindof, the `useprivate` is adding an unnecessary layer I think, you're not reading from `params`

Comment (ttodua) [reply ts/src/coinbase.ts:347]: but what should be the meaningful name so users didn't ask "what does this option mean"?

Comment (ttodua) [reply ts/src/coinbase.ts:347]: i've marked this ready for review, please feel free to edit that name inline as you prefer, otherwise it's ready

**Comment by jtashiro** [comment]
Thank guys -I've pulled this version locally, installed it and tested the 'usePrivate' option.  Looks good - am not getting those rate limit issues.

**Pull request merged**